### ISSUE TITLE
Permission RECEIVE_SMS declared but not used

### ISSUE THREAD

**Issue by denis-bogdanas** [Freeyourgadget/Gadgetbridge#501]
Hello, I'm Denis Bogdanas, a research assistant at Oregon Stat University.
I'm working on a tool that automatically introduces runtime permission checks and requests for Android 6 apps. As part of this study I analyzed this app. I was looking at the way permissions are referred in the code.

This app declares permission <RECEIVE_SMS> in the apk manifest, but doesn't have any code that requires them. It is possible that source code doesn't declare the permissions. I only analyzed apk builds from f-droid. Yet android build mechanism may introduce them when the app uses certain libraries:

https://commonsware.com/blog/2015/06/25/hey-where-did-these-permissions-come-from.html

It won't create any problems on Android 6, but on Android 5- users may wonder why the app needs these permissions and be skeptical of installing the app.

Do you think this is an issue?

best regards,
Denis

**Comment by ashimokawa** [comment]
@denis-bogdanas 

Thanks for the report.

Unfortunately your findings are wrong and we DO USE the RECEIVE_SMS permission.

Maybe you missed it because we start receivers at runtime. In our case the code is in 

app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/SMSReceiver.java

and an instance of SMSReceiver is created in

app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/DeviceCommunicationService.java

`       
if (mSMSReceiver == null) {
                mSMSReceiver = new SMSReceiver();
                registerReceiver(mSMSReceiver, new IntentFilter("android.provider.Telephony.SMS_RECEIVED"));
            }
`

**Comment by denis-bogdanas** [comment]
Thanks a lot for reply! You helped me realize all the mess Google did with javadoc and permissions in particular. 

I mined the permission specification from xml annotations coming with Android SDK and from javadoc comments from Android source code. Then I got some more permission mappings by analyzing how apps use them. For SMS_RECEIVED_ACTION there's no text mentioning required permission in javadoc, but when you press Ctrl+Q on this field, some extra text appears at the end saying "Requires RECEIVE_SMS to receive.". Crazy. That's why my tools couldn't detect the permission properly.

There's another thing that puzzles me. This app declares RECEIVE_SMS in manifest but requests READ_SMS at runtime. Are these two interchangeable?
